acq4/modules/AutomationDebug.py

```diff
@@ -126,27 +138,98 @@ def doPipetteCalibrationTest(self, _future):
                 calibrtion_fut = calibratePipette(pipette, camera, camera.scopeDev)
                 _future.waitFor(calibrtion_fut)
                 error = np.linalg.norm(pipette.globalPosition() - true_tip_position)
-                self._pipetteLog.append(f'Calibration complete: {error*1e6:.2g}µm error')
+                self.sigLogMessage.emit(f"Calibration complete: {error*1e6:.2g}µm error")
                 if error > 50e-6:
                     self.failedCalibrations.append(error)
                     i = len(self.failedCalibrations) - 1
-                    self._pipetteLog.append(f'....so bad. Why? Check man.getModule("AutomationDebug").failedCalibrations[{i}]')
+                    self.sigLogMessage.emit(
+                        f'....so bad. Why? Check man.getModule("AutomationDebug").failedCalibrations[{i}]'
+                    )
 
             except Exception as e:
                 if self._testing_pipette:
                     raise
-                self._pipetteLog.append('Calibration interrupted by user request')
+                self.sigLogMessage.emit("Calibration interrupted by user request")
+
+    def trackFeatures(self):
+        if self._trackingFeatures:
+            self._trackingFeatures = False
+            self._trackFeaturesBtn.setText("Track target by features")
+            self._trackFeaturesBtn.setStyleSheet("")
+            self._featureTrackingFuture.stop()  # TODO force kill the thread after a timeout
+        else:
+            self._trackingFeatures = True
+            self._trackFeaturesBtn.setText("Stop tracking")
+            self._trackFeaturesBtn.setStyleSheet("QPushButton {background-color: green; color: white}")
+            self._featureTrackingFuture = self.doFeatureTracking()
+            self._featureTrackingFuture.sigFinished.connect(self._handleFeatureTrackingFinish)
+
+    @future_wrap
+    def doFeatureTracking(self, _future: Future):
+        _future.sleep(0.1)
+        from acq4.util.visual_tracker import PyrLK3DTracker, ObjectStack, ImageStack
```

Comment:
we can swap out for CellPoseTracker with ease, and I want to do live tests with both